sites/website/src/docs/apps-and-experiences/dependency-injection.md

```diff
@@ -0,0 +1,181 @@
+- How do we manage memory and control the lifetimes of objects in such a system?
+- How does our user interface, often controlled by a different engine, gain access to our composed system capabilities?
+
+To address these challenges, FAST introduces the concept of a *dependency injection container*. A DI Container is a sub-system with the responsibilities of understanding dependency relationships, constructing objects with their dependencies, delivering dependencies to components, and managing lifetimes.
+
+## Creating a DI Container
+
+DI Containers can exist in a hierarchy, allowing child containers to override the dependencies of the parent, but typically there is one root container in which all the system's services are registered. In a typical FAST application, you will want this container to be associated with the `document.body` so that all UI component children can gain access to its capabilities. Here's how you would create that root container:
+
+```ts
+const container = DI.getOrCreateDOMContainer();
+```
+
+You'll want to create and configure your root container as early as possible in your application lifecycle, typically in your application's entry point module. If you are using FAST's Design System features or its components, we've integrated them with DI, so that you can figure everything with a unified API. Instead of calling `DI.getOrCreateDOMContainer()` you can simply import the Design System Provider function and use that. Here's some code that you may have seen in other parts of our documentation, that does just that:
+
+```ts
+provideFASTDesignSystem()
+    .register(
+        fastButton()
+    );
+```
+
+The `register` method of the `DesignSystem` actually delegates directly to the DI container. As a result, you can register your Web Components and all your application dependencies in the same place, following the same patterns.
+
+### Example Scenario
+
+Once you have a container, you can use it to register and retrieve system dependencies. This is best demonstrated with a typical scenario. Imagine that we have a `NewAccountScreen` web component in our app. Users navigate to this screen to create a new account. This screen is dependent on an `AccountService`, which is able to create accounts and login new users. The `AccountService` itself is dependent on a `UserSession` and an `HTTPClient`. The `HTTPClient` is dependent on a `ServiceBaseURL` string. Let's set this up, working from the inside out.
+
+## Creating DI Keys
+
+The FAST DI can handle any kind of dependency, including primitive values like strings. So, we can register a `ServiceBaseURL` with our container and any HTTP service that needs it can simply request it by key. This is nice because it lets us centralize this configuration, making it easy to swap out in different environments. Here's how you define a strongly typed *key* that symbolizes an interface to a dependency.
```

Comment:
```suggestion
The FAST DI can handle any kind of dependency, including primitive values like strings. Simply register a `ServiceBaseURL` with the container and any HTTP service that needs it can simply request it by key. This allows for a centralized configuration, making it easy to swap out in different environments. Here's how you define a strongly typed *key* that symbolizes an interface to a dependency.
```